web/vibe/web/internal/rest/jsclient.d

```diff
@@ -100,12 +105,24 @@ class JSRestClientSettings
 			// extract the server part of the URL
 			auto burl = URL(intf.baseURL);
 			burl.pathString = "/";
-			fout.serializeToJson(burl.toString()[0 .. $-1]);
-			// and then assemble the full path piece-wise
-			foreach (p; route.fullPathParts) {
-				fout.put(" + ");
-				if (!p.isParameter) fout.serializeToJson(p.text);
-				else fout.formattedWrite("toRestString(%s)", p.text);
+
+			// if url not empty
+			if (intf.baseURL != intf.basePath) {
+				fout.serializeToJson(burl.toString()[0 .. $-1]);
+				// and then assemble the full path piece-wise
+				foreach (p; route.fullPathParts) {
+					fout.put(" + ");
+					fmtParam(fout, p);
+				}
+			} else {
+				if (route.fullPathParts.length)
+				{
+					fmtParam(fout, route.fullPathParts[0]);
+					foreach (p; route.fullPathParts[1..$]) {
+						fout.put(" + ");
+						fmtParam(fout, p);
+					}
+				}
```

Comment:
Would be good to have an `assert(false, "Empty route function path");` here in the else case, so that no invalid JavaScript can be produced here. (the `else if` could also be merged to a single line without an extra block)